tests/e2e/normalize.go

```diff
@@ -95,9 +95,18 @@ func normalizeKRMObject(u *unstructured.Unstructured, project testgcp.GCPProject
 	visitor.stringTransforms = append(visitor.stringTransforms, func(path string, s string) string {
 		return strings.ReplaceAll(s, project.ProjectID, "${projectId}")
 	})
+
+	// Update the project number with variable marker.
+	if u.GetKind() == "Project" {
+		// The ProjectNumber is ProjectID based. For those tests relying on the Mock Resource Manager server,
+		// the ProjectID is dynamic, different from the default mock test Project "mock-project",
+		// so the number is different as well.
+		visitor.replacePaths[".status.number"] = fmt.Sprintf("%d", project.ProjectNumber)
```

Comment:
Maybe we should replace with `${projectNumber}` ?  Though TBH I'm not entirely sure why it doesn't get replaced below.  Maybe the order in which these replacements run?

Reply:
Yes, yes, I also want to use `${projectNumber}`. Please see the last commit. 
The reason of not replacing is that those tests have two Projects, one is the "mock-project", the other is the real project which `uniqId` is dynamically generated. The below replacement `strings.ReplaceAll(s, fmt.Sprintf("%d", project.ProjectNumber), "${projectNumber}")` can only find the `mock-project` number, but not the dynamic ones. This only happens for tests that call the mock resource manager.